flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/executor/JdbcBatchStatementExecutor.java

```diff
@@ -16,29 +16,27 @@
  * limitations under the License.
  */
 
-package org.apache.flink.api.java.io.jdbc.writer;
+package org.apache.flink.api.java.io.jdbc.executor;
 
-import org.apache.flink.api.java.tuple.Tuple2;
-import org.apache.flink.types.Row;
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.java.io.jdbc.JdbcStatementBuilder;
 
-import java.io.Serializable;
 import java.sql.Connection;
 import java.sql.SQLException;
+import java.util.function.Function;
 
 /**
- * JDBCWriter used to execute statements (e.g. INSERT, UPSERT, DELETE).
+ * Executes the given JDBC statement in batch for the accumulated records.
  */
-public interface JDBCWriter extends Serializable {
+@Internal
+public interface JdbcBatchStatementExecutor<T> {
```

Comment:
Here all the methods are expected to only throw `SQLException` and then, in many cases, we wrap the exceptions in other types. Does it make sense to simply have these methods throw an `Exception`?

Reply:
On higher levels, it is wrapped because there the allowed exception is `IOException`, not `SQLException`. So we can only declare `throws IOException` without wrapping.

But IMO `SQLException` is better here because:
- caller can actually do something with it
- it has an appropriate abstraction level

Reply:
After further looking at code, I noticed it was wrapped in one place into `RuntimeException` instead of `IOException` so I fixed wrapping.